x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/analytics/nested_top_metrics_sort.yml

```diff
@@ -278,3 +396,653 @@ setup:
+                        size: 1
+                        sort:
+                          order_date: desc
+
+  - match: { hits.total.value: 16 }
+  - match: { hits.total.relation: "eq" }
+  - length: { aggregations.name.buckets: 5 }
+
+  - match: { aggregations.name.buckets.0.key: "alice" }
+  - match: { aggregations.name.buckets.0.doc_count: 2 }
+  - match: { aggregations.name.buckets.0.field_exists.doc_count: 2 }
+  - length: { aggregations.name.buckets.0.field_exists.top_metrics.top: 1 }
+  - length: { aggregations.name.buckets.0.field_exists.top_metrics.top.0.sort: 1 }
+  - match: { aggregations.name.buckets.0.field_exists.top_metrics.top.0.sort.0: "2021-09-09T17:50:00.000Z" }
+  - match: { aggregations.name.buckets.0.field_exists.top_metrics.top.0.metrics.version: 1 }
+
+  - match: { aggregations.name.buckets.1.key: "bob" }
+  - match: { aggregations.name.buckets.1.doc_count: 1 }
+  - match: { aggregations.name.buckets.1.field_exists.doc_count: 1 }
+  - length: { aggregations.name.buckets.1.field_exists.top_metrics.top: 1 }
+  - length: { aggregations.name.buckets.1.field_exists.top_metrics.top.0.sort: 1 }
+  - match: { aggregations.name.buckets.1.field_exists.top_metrics.top.0.sort.0: "2021-10-12T18:39:00.000Z" }
+  - match: { aggregations.name.buckets.1.field_exists.top_metrics.top.0.metrics.version: 1 }
+
+  - match: { aggregations.name.buckets.2.key: "bruce" }
+  - match: { aggregations.name.buckets.2.doc_count: 1 }
+  - match: { aggregations.name.buckets.2.field_exists.doc_count: 1 }
+  - length: { aggregations.name.buckets.2.field_exists.top_metrics.top: 1 }
+  - length: { aggregations.name.buckets.2.field_exists.top_metrics.top.0.sort: 1 }
+  - match: { aggregations.name.buckets.2.field_exists.top_metrics.top.0.sort.0: "2021-08-29T18:47:00.000Z" }
+  - match: { aggregations.name.buckets.2.field_exists.top_metrics.top.0.metrics.version: 1 }
+
+  - match: { aggregations.name.buckets.3.key: "charlie" }
+  - match: { aggregations.name.buckets.3.doc_count: 3 }
+  - match: { aggregations.name.buckets.3.field_exists.doc_count: 3 }
+  - length: { aggregations.name.buckets.3.field_exists.top_metrics.top: 1 }
+  - length: { aggregations.name.buckets.3.field_exists.top_metrics.top.0.sort: 1 }
+  - match: { aggregations.name.buckets.3.field_exists.top_metrics.top.0.sort.0: "2021-12-10T19:33:00.000Z" }
+  - match: { aggregations.name.buckets.3.field_exists.top_metrics.top.0.metrics.version: 1 }
+
+  - match: { aggregations.name.buckets.4.key: "daniel" }
+  - match: { aggregations.name.buckets.4.doc_count: 3 }
+  - match: { aggregations.name.buckets.4.field_exists.doc_count: 3 }
+  - length: { aggregations.name.buckets.4.field_exists.top_metrics.top: 1 }
+  - length: { aggregations.name.buckets.4.field_exists.top_metrics.top.0.sort: 1 }
+  - match: { aggregations.name.buckets.4.field_exists.top_metrics.top.0.sort.0: "2021-11-09T17:29:00.000Z" }
+  - match: { aggregations.name.buckets.4.field_exists.top_metrics.top.0.metrics.version: 1 }
+
+---
```

Comment:
The last two tests are here just to highlight that the behaviour of the top_metric aggregation changes with the value of `rewrite_to_filter_by_filter`. This is not correct, we would like to have consistent behaviour. I will submit an issue to track this and point the issue to these two tests (and their counterpart where the setting is true).

Reply:
I will create the issue after merging this PR so that I can reference the two tests in the ticket.